### PR TITLE
planner: add documents for cost model variables

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@qw4990 Could you please invite a tech review? Thanks

File: cost-model.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+>
+> - Cost Model Version 2 目前为实验特性，不建议在生产环境中使用。
+>
+> - 切换代价模型版本可能引起计划的变动。
+
+在 v6.2 版本中，我们引入了新的代价模型 Cost Model Version 2，在内部测试中，它比旧版代价模型更加准确。
+
+如下图，右边新版的代价模型估算出来的代价和计划实际的执行时间更加吻合：
+
+![cost-model-ver2](/media/cost-model-ver2.png)
```
Comment (time-and-fate) [inline cost-model.md:53]: 感觉可以不用贴实验结果，这个实验结果本身对用户也不太容易理解。我觉得只需要介绍说内部测试表明它更准确就好了。

File: cost-model.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+在处理查询 `select * from t where b<100 and c<100` 时，假设 TiDB 对 `b<100` 和 `c<100` 的行数估计分别为 20 和 500，`INT` 类型索引行宽为 8，则 TiDB 会分别计算两个索引的代价：
+
++ 索引 `b` 的扫描代价 = `b<100 行数` * `索引 b 行宽` = 20 * 8 = 160
++ 索引 `c` 的扫描代价 = `c<100 行数` * `索引 c 行宽` = 500 * 8 = 4000
+
+由于扫描 `b` 的代价更低，因此 TiDB 会选择 `b` 索引。
+
+上述是一个简化后的例子，只是用于做原理解释，实际 TiDB 的代价模型会更加复杂，具体的代价公式可以见[源码](https://github.com/pingcap/tidb/blob/master/planner/core/plan_cost.go)。
```
Comment (time-and-fate) [inline cost-model.md:39]: 这个链接总是链接到 master 分支的源码的，但是文档是分版本的，感觉后面维护可能会有问题。
我觉得可以不用贴源码的链接。

File: cost-model.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+>
+> - 切换代价模型版本可能引起计划的变动。
+
+在 v6.2 版本中，我们引入了新的代价模型 Cost Model Version 2，在内部测试中，它比旧版代价模型更加准确。
+
+如下图，右边新版的代价模型估算出来的代价和计划实际的执行时间更加吻合：
+
+![cost-model-ver2](/media/cost-model-ver2.png)
+
+你可以通过设置变量 `tidb_cost_model_version` 来控制代价模型的版本。
```
Comment (time-and-fate) [inline cost-model.md:55]: 可以加一个到系统变量页面这个变量对应位置的链接。

File: cost-model.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
+title: 代价模型
```
Comment (Oreoxmt) [inline cost-model.md:2]: 在 TOC.md 中添加 `- [代价模型](/cost-model.md)`
```
      - 物理优化
        - [物理优化概览](/sql-physical-optimization.md)
        - [索引的选择](/choose-index.md)
        - [统计信息简介](/statistics.md)
        - [错误索引的解决方案](/wrong-index-solution.md)
        - [Distinct 优化](/agg-distinct-optimization.md)
        - [代价模型](/cost-model.md)
```

File: cost-model.md (hunk 1 above)
Comment (qw4990) [reply cost-model.md]: 已删除

File: cost-model.md (hunk 2 above)
Comment (qw4990) [reply cost-model.md:39]: 已删除

File: cost-model.md (hunk 3 above)
Comment (qw4990) [reply cost-model.md]: 已更新

File: cost-model.md (hunk 4 above)
Comment (qw4990) [reply cost-model.md:2]: 已更新

**Review by time-and-fate — APPROVED** [review]

File: cost-model.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+aliases: ['/docs-cn/dev/cost-model/']
```
Comment (TomShawn) [inline cost-model.md:3]: You can also add a summary.
```suggestion
```

File: cost-model.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+# 代价模型
+
+如下图，TiDB 在物理优化时会使用代价模型来进行索引选择和算子选择：
```
Comment (TomShawn) [inline cost-model.md:8]:
```suggestion
TiDB 在进行[物理优化](/sql-physical-optimization.md)时会使用代价模型来进行索引选择和算子选择，如下图所示：
```

File: system-variables.md [hunk 7]
```diff
@@ -548,6 +548,21 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+> **警告：**
+>
+> - 当前 Cost Model Version 2 为实验特性，不建议在生产环境中使用。
+> - 切换代价模型版本可能会引起查询计划的变动。
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`1`
+- 可选值：`[1, 2]`
+- TiDB v6.2.0 引入了新版本的代价模型 [Cost Model Version 2](/cost-model.md#cost-model-version-2)，在内部测试中比现有老版本代价模型更加准确。
```
Comment (TomShawn) [inline system-variables.md:562]:
```suggestion
- TiDB v6.2.0 引入了代价模型 [Cost Model Version 2](/cost-model.md#cost-model-version-2)，在内部测试中比此前版本的代价模型更加准确。
```

File: system-variables.md [hunk 8]
```diff
@@ -548,6 +548,21 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 通过将 `tidb_cost_model_version` 设置为 `2` 可以启用新版本的代价模型，设置为 `1` 则继续使用老版本代价模型。
```
Comment (TomShawn) [inline system-variables.md:563]: 文档中可尽量避免「新旧」这类相对的词语。相对未来某个版本，有了 Cost Model Version 3 后，现在的 Cost Model Version 2 就成了「旧版本」了。
```suggestion
- 通过将 `tidb_cost_model_version` 设置为 `2` 可以启用 Cost Model Version 2 代价模型，设置为 `1` 则继续使用 Cost Model Version 1 代价模型。
```

File: system-variables.md [hunk 9]
```diff
@@ -1400,6 +1415,15 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
 > * PHP [mysqli](https://dev.mysql.com/doc/apis-php/en/apis-php-mysqli.quickstart.multiple-statement.html) (`mysqli_multi_query`)
 
+### `tidb_enable_new_cost_interface` <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 可选值：`OFF`，`ON`
+- TiDB v6.2.0 对代价模型的实现进行了代码层面的重构，这个变量用来控制是否使用重构后的代价模型 [Cost Model Version 2](/cost-model.md#cost-model-version-2)。
+- 重构后的代价模型使用完全一样的代价公式，因此不会引起计划选择的变动，此开关默认打开。
```
Comment (TomShawn) [inline system-variables.md:1425]: 这里是否也可以提一下「对代价公式进行了更精确的回归校准，调整了部分代价公式，比旧版代价公式更加准确」这个信息呢？

File: cost-model.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+---
+title: 代价模型
```
Comment (Oreoxmt) [inline cost-model.md:2]:
```suggestion
title: 代价模型
summary: 介绍代价模型的原理和 Cost Model Version 2。
```

File: system-variables.md (hunk 9 above)
Comment (qw4990) [reply system-variables.md:1425]: 哦这里不用提，这个开关不涉及公式的变动；

File: cost-model.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+summary: 介绍 TiDB 进行物理优化时所使用的代价模型的原理和 Cost Model Version 2。
```
Comment (Oreoxmt) [inline cost-model.md:3]:
```suggestion
summary: 介绍 TiDB 进行物理优化时所使用的代价模型的原理。
```

File: cost-model.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  `b` int(11) DEFAULT NULL,
+  `c` int(11) DEFAULT NULL,
+  KEY `b` (`b`),
+  KEY `c` (`c`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+在处理查询 `SELECT * FROM t WHERE b<100 and c<100` 时，假设 TiDB 对 `b<100` 和 `c<100` 的行数估计分别为 20 和 500，`INT` 类型索引行宽为 8，则 TiDB 会分别计算两个索引的代价：
```
Comment (Oreoxmt) [inline cost-model.md:32]:
```suggestion
在处理查询 `SELECT * FROM t WHERE b < 100 and c < 100` 时，假设 TiDB 对 `b < 100` 和 `c < 100` 的行数估计分别为 20 和 500，`INT` 类型索引行宽为 8，则 TiDB 会分别计算两个索引的代价：
```

File: cost-model.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
++ 索引 `b` 的扫描代价 = `b<100 行数` * `索引 b 行宽` = 20 * 8 = 160
```
Comment (Oreoxmt) [inline cost-model.md:34]:
```suggestion
+ 索引 `b` 的扫描代价 = `b < 100 行数` * `索引 b 行宽` = 20 * 8 = 160
```

File: cost-model.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
++ 索引 `c` 的扫描代价 = `c<100 行数` * `索引 c 行宽` = 500 * 8 = 4000
```
Comment (Oreoxmt) [inline cost-model.md:35]:
```suggestion
+ 索引 `c` 的扫描代价 = `c < 100 行数` * `索引 c 行宽` = 500 * 8 = 4000
```

File: system-variables.md [hunk 15]
```diff
@@ -1400,6 +1415,16 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+- 从 v6.1.0 升级上来的用户此开关可能处于 `OFF` 状态，可以不用去修改它。
```
Comment (TomShawn) [inline system-variables.md:1426]:
```suggestion
- 从 v6.2.0 及之前版本升级至 v6.2.0 或之后版本的用户，此开关保持原来的 `OFF` 状态，用户无需修改。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**